### PR TITLE
Give the ability to create new postings, applications

### PR DESCRIPTION
**Pull request by ramy-att**
- Create postings #23 
- Create applications #23 
- Changed all PUT to PATCH so that we can update only the required field (@khaled106 this should be changed in the edit profile request)
- Changed all the employer attributes to match the convention (employerName, ...)  as this will facilitate managing who is who when creating, editing and deleting postings and applications - #93 

I will work now on allowing the employer to view all of his postings and applications and allowing the student to view his own applications in #95

### CODE REVIEW

File: server/controller/application-controller.js [hunk 1]
```diff
@@ -36,12 +34,11 @@ const addApplication = async (req, res, next) => {
     // defining an application
     application = new Application({
       studentID, // = getStudentID()
-      studentResume,
-      studentCoverLetter,
-      extraDetailsForEmployerByStudent,
+      postingID,
+      extraDetails,
```
Comment (TarekElalfi) [inline server/controller/application-controller.js:38]: Do we still need the extra details category ? I think we can keep it and use it as a feature.

File: server/controller/employer-controller.js [hunk 2]
```diff
@@ -16,46 +16,40 @@ const getAllEmployers = async (req, res, next) => {
...
     return res.status(400).json({ err: "User already exists" });
   } else if (
-    !name ||
-    name.trim() === "" ||
-    !email ||
-    email.trim() === "" ||
+    !employerName ||
+    employerName.trim() === "" ||
+    !employerEmail ||
+    employerEmail.trim() === "" ||
```
Comment (TarekElalfi) [inline server/controller/employer-controller.js:33]: This change will make the code clearer thanks for refactoring Ramy!

File: server/routes/application-routes.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ const applicationRouter = express.Router(); // contains all the request methods(
 
 applicationRouter.get("/", getAllApplications); //get
 applicationRouter.post("/", addApplication); //create
-applicationRouter.put("/:id", updateApplication); //update
-applicationRouter.delete("/:id", deleteApplication);//only accessable to 
+applicationRouter.patch("/:id", updateApplication); //update
+applicationRouter.delete("/:id", deleteApplication); //only accessable to
```
Comment (TarekElalfi) [inline server/routes/application-routes.js:16]: That would make it easier for a change thanks

**Review by TarekElalfi — APPROVED** [review]
Great work keep it up

File: server/controller/application-controller.js (hunk 1 above)
Comment (ramy-att) [reply server/controller/application-controller.js:38]: @TarekElalfi I kept the extraDetails which would be a text passed in by the student based on @MazennW's original implementation

**Pull request merged**